torchtitan/components/optimizer.py

```diff
@@ -349,25 +353,67 @@ def _update_expert_bias(
         )
         # TODO: Currently this sync is blocking (thus exposed) and happens on the
         # default compute stream. Need to assess if this is OK performance-wise.
+        tokens_per_expert_list = []
         for model_part in model_parts:
             for transformer_block in model_part.layers.values():
-                if transformer_block.moe_enabled:
+                if not transformer_block.moe_enabled:
+                    continue
+                moe = transformer_block.moe
+                if moe.load_balance_coeff is None:
+                    continue
```

Comment:
```suggestion
                if transformer_block.moe.load_balance_coeff is None:
                    return
```
we should just return, as this setting is uniformly enabled / disabled across all MoE layers